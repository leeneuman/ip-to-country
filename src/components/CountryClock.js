import React, { useEffect, useState } from 'react';
import { getTimezone } from '../api/IpApi';
import { isValidIP } from '../helpers/dataHelpers';
import Clock from 'react-live-clock';

function CountryClock({ ip, show = false }) {
	const [ timeZone, setTimeZone ] = useState('');

	useEffect(
		() => {
			if (isValidIP(ip)) {
				getTimezone(ip).then((res) => setTimeZone(res.data.timezone)).catch((err) => console.log(err.message));
			}
		},
		[ ip ]
	);

	if (!timeZone || !show) return null;

	return <Clock format={'HH:mm:ss'} ticking={true} timezone={timeZone} />;
}

export default CountryClock;
