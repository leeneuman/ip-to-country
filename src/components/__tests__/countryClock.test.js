import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CountryClock from '../countryClock';

afterEach(cleanup);

test('should render clock component', () => {
	render(<CountryClock ip={'79.179.206.105'} show={true} />);
	const clockElement = screen.getByTestId('clock');
	expect(clockElement).toBeInTheDocument();
});

test('matches snapshot', () => {
	const tree = renderer.create(<CountryClock ip={'79.179.206.105'} show={true} />).toJSON();
	expect(tree).toMatchSnapshot();
});
