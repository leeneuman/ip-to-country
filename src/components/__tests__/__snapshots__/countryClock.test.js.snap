// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="clock"
>
  <time
    dateTime={"2021-04-30T09:32:23.625Z"}
  >
    12:32:23
  </time>
</div>
`;
